### ISSUE TITLE
y4m files from xiph.org cannot be converted to animated AVIF

### ISSUE THREAD

**Issue by FrankGalligan** [AOMediaCodec/libavif#554]
I tried encoding a few y4m files from https://media.xiph.org/video/derf/

But I get errors like this:
Failed to parse y4m header (not enough information): /ice_qcif_15fps.y4m
Cannot determine input file format:

If I run this command in FFmpeg:
ffmpeg -i ./ice_qcif_15fps.y4m -pix_fmt yuv420p -f yuv4mpegpipe ./ice_qcif_15fps_pass_thru.y4m

Then run avifenc:
./avifenc ./ice_qcif_15fps_pass_thru.y4m ./animated.avif

This works

**Comment by joedrago** [comment]
Can you supply the text header here from the file that doesn't work? I'm sure we're just missing some format that we can probably map pretty easily.

**Comment by FrankGalligan** [comment]
Yeah so it looks like the header in the original files from  https://media.xiph.org/video/derf/ has:
`YUV4MPEG2 W176 H144 F15:1 Ip A128:117`

When you run it through FFmpeg the header is:
`YUV4MPEG2 W176 H144 F15:1 Ip A128:117 C420jpeg XYSCSS=420JPEG`

Other than that the files look the same.

**Comment by joedrago** [comment]
Missing the color space (`C`) from the header is interesting. Is there a default color space for y4m that I don't know about? I'd be _really_ surprised if it was C420jpeg (as I believe that is full range). I could see it being `C420` though.

**Comment by tongyuantongyu** [comment]
`C420jpeg` is the default, but it's only saying YUV420 with chroma position at center. Color range is signaled through `XCOLORRANGE` tag.

The following code from [mjpegtools](https://sourceforge.net/projects/mjpeg/files/mjpegtools/2.2.0/) yuv4mpeg.c/utils/yuv4mpeg.c#L624 confirms this.
```c
  /* Without 'C' tag or any other chroma spec, default to 420jpeg */
  if (i->chroma == Y4M_UNKNOWN) 
    i->chroma = Y4M_CHROMA_420JPEG;
```

**Comment by joedrago** [comment]
@FrankGalligan Can you try this locally and see if the behavior works for you? Change line 212 of `y4m.c` from:

    frame.format = AVIF_PIXEL_FORMAT_NONE;

to

    frame.format = AVIF_PIXEL_FORMAT_YUV420; // If a y4m does not contain a 'C' tag, it defaults to 420jpeg

If @tongyuantongyu (and mjpegtools) is correct and everyone considers y4m's "default" to be 420jpeg, this should provide that.

**Comment by tongyuantongyu** [comment]
This is also ffmpeg's behavior:
https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/yuv4mpegenc.c#L111
